pandas/sparse/scipy_sparse.py

```diff
@@ -0,0 +1,102 @@
+"""
+Interaction with scipy.sparse matrices.
+
+Currently only includes SparseSeries.to_coo helpers.
+"""
+from pandas.core.frame import DataFrame
+from pandas.core.index import MultiIndex, Index
+from pandas.core.series import Series
+import itertools
+import numpy
+from pandas.compat import OrderedDict
+from pandas.tools.util import cartesian_product
+
+
+def _get_label_to_i_dict(labels, sort_labels=False):
+    """ Return OrderedDict of unique labels to number. Optionally sort by label. """
+    labels = Index(map(tuple, labels)).unique().tolist()  # squish
+    if sort_labels:
+        labels = sorted(list(labels))
+    d = OrderedDict((k, i) for i, k in enumerate(labels))
+    return(d)
+
+
+def _get_index_subset_to_coord_dict(index, subset, sort_labels=False):
+    ilabels = list(zip(*[index.get_level_values(i) for i in subset]))
+    labels_to_i = _get_label_to_i_dict(ilabels, sort_labels=sort_labels)
+    return(labels_to_i)
+
+
+def _check_is_partition(parts, whole):
+    whole = set(whole)
+    parts = [set(x) for x in parts]
+    if set.intersection(*parts) != set():
+        raise ValueError(
+            'Is not a partition because intersection is not null.')
+    if set.union(*parts) != whole:
+        raise ValueError('Is not a partition becuase union is not the whole.')
+
+
+def _to_ijv(ss, ilevels=(0,), jlevels=(1,), sort_labels=False):
+    """ For arbitrary (MultiIndexed) SparseSeries return (v, i, j, ilabels, jlabels) where (v, (i, j)) is suitable for
+    passing to scipy.sparse.coo constructor. """
+    # index and column levels must be a partition of the index
+    _check_is_partition([ilevels, jlevels], range(ss.index.nlevels))
+
+    # from the SparseSeries: get the labels and data for non-null entries
+    values = ss._data.values._valid_sp_values
+    blocs = ss._data.values.sp_index.blocs
+    blength = ss._data.values.sp_index.blengths
+    nonnull_labels = list(
+        itertools.chain(*[ss.index.values[i:(i + j)] for i, j in zip(blocs, blength)]))
+
+    def get_indexers(levels):
+        """ Return sparse coords and dense labels for subset levels """
+        values_ilabels = [tuple(x[i] for i in levels) for x in nonnull_labels]
+        labels_to_i = _get_index_subset_to_coord_dict(
+            ss.index, levels, sort_labels=sort_labels)
+        i_coord = [labels_to_i[i] for i in values_ilabels]
```

Comment:
If `labels_to_i` is a MultiIndex instead of an OrderedDict (like I recommend above), then this is as simple as:
`i_coord = labels_to_i.get_indexer(values_ilabels)` -- and again, we're using the fast pandas methods, not a loop in pure Python. The only cavaet of using `get_indexer` is that missing values are marked by `-1` instead of raising `KeyError`, so you might need to check for that explicitly (if you haven't already guaranteed that can't happen).